ios/ScannerBle.swift

```diff
@@ -18,50 +18,22 @@
 import CoreBluetooth
 
 ///
-/// BeaconScannerDelegate
-///
-/// Implement this to receive notifications about beacons.
-protocol BeaconScannerDelegate {
-  func urlContextChanged(beaconScanner: BeaconScanner)
-}
-
-///
 /// BeaconScanner
 ///
 /// Scans for Eddystone compliant beacons using Core Bluetooth. To receive notifications of any
 /// sighted beacons, be sure to implement BeaconScannerDelegate and set that on the scanner.
 ///
-class BeaconScanner: NSObject, CBCentralManagerDelegate {
-  
-  var delegate: BeaconScannerDelegate?
-  
+class BeaconScanner: NSObject, Scanner, CBCentralManagerDelegate {
   private var centralManager: CBCentralManager!
   private let beaconOperationsQueue: dispatch_queue_t = dispatch_queue_create("beacon_operations_queue", nil)
   private var shouldBeScanning: Bool = false
-  private var _urlFound = Set<NSURL>()
-  
-  var urlFound: Set<NSURL> {
-    get {
-      return _urlFound
-    }
-  }
-  
-  var urls: Array<String> {
-    get {
-      var urls = Set<String>();
-      for url in _urlFound {
-        urls.insert("\(url)");
-      }
-      return Array(urls);
-    }
-  }
+  private var callback: ((Dictionary<String, AnyObject>) -> Void)!
   
-  override init() {
+  init(callback: (Dictionary<String, AnyObject>) -> Void) {
```

Comment:
I did some research into this and came to the conclusion that callbacks are cleaner and more modern and delegates were an older pattern. Here's a [blog post](https://medium.com/cobe-mobile/why-you-shouldn-t-use-delegates-in-swift-7ef808a7f16b#.o5mhn7l5w) for more info.

Also [here's an example](https://developer.apple.com/library/prerelease/iOS/documentation/UIKit/Reference/UIAlertAction_Class/index.html#//apple_ref/occ/clm/UIAlertAction/actionWithTitle:style:handler:) of Apple using callbacks in a newer core library.